### PR TITLE
ci: Deploy a new Lambda layer on Agent release

### PR DESCRIPTION
**Pull request by chynesNR**
When we deploy a new version of the .NET Agent, this will create a tag in the [New Relic Lambda Layers repo](https://github.com/newrelic/newrelic-lambda-layers/), which in turn will trigger creation of a new .NET Lambda layer. The tag needs to be in the format `vX.Y.Z_dotnet`. This PR is waiting for [that workflow](https://github.com/newrelic/newrelic-lambda-layers/pull/228) to be merged.

### CODE REVIEW

File: .github/workflows/deploy_agent.yml [hunk 1]
```diff
@@ -443,12 +448,23 @@ jobs:
       run_id: ${{ needs.get-release-info.outputs.workflow_run_id }}
     secrets: inherit
 
+  deploy-lambda-layer:
+    permissions:
+      contents: write
```
Comment (nrcventura) [inline .github/workflows/deploy_agent.yml:453]: Do we need to move this permission into the job that is ultimately called? Or do we not need write access?

Comment (chynesNR) [reply .github/workflows/deploy_agent.yml]: Yeah I think you're right, since the tag isn't getting created here.

**Review by nrcventura — APPROVED** [review]

**Review by nr-ahemsath — APPROVED** [review]
Looks correct to me! 👍

**Comment by chynesNR** [comment]
Putting this aside for now. We're going to try using the [auto-layer](https://source.datanerd.us/aws-lambda/auto-layer-releases/pull/8).

**Pull request closed without merging**